tools/hsmtools.c

```diff
@@ -44,25 +53,33 @@ static bool ensure_hsm_secret_exists(int fd)
 	return true;
 }
 
-static int decrypt_hsm(const char *hsm_secret_path, const char *passwd)
+static void get_hsm_secret(struct secret *hsm_secret,
+                           const char *hsm_secret_path)
 {
 	int fd;
-	struct stat st;
-	struct secret key, hsm_secret;
+
+	fd = open(hsm_secret_path, O_RDONLY);
+	if (fd < 0)
+		errx(ERROR_HSM_FILE, "Could not open hsm_secret");
```

Comment:
Use err() not errx() here, which will tell them what the error was?

Reply:
err.. I confused the err functions